internal/serve/middleware/middleware.go

```diff
@@ -249,6 +249,28 @@ func TenantMiddleware(tenantManager tenant.ManagerInterface, authManager auth.Au
 	}
 }
 
+func BasicAuthMiddleware(adminAccount, adminApiKey string) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			ctx := req.Context()
+
+			if adminAccount == "" || adminApiKey == "" {
+				httperror.InternalError(ctx, "Admin account and API key are not set", nil, nil).Render(rw)
+				return
+			}
+
+			accountUserName, apiKey, ok := req.BasicAuth()
```

Comment:
Started by implementing the headers manually but then found this method. it performs the decoding and follows the `RFC 2617, Section 2` properly.